From 45de32c25c553adc1463e2b86617d5a7f7266a45 Mon Sep 17 00:00:00 2001
From: Othmane AHL ZOUAOUI <othmane.ahlzouaoui@st.com>
Date: Wed, 20 Mar 2024 15:59:12 +0100
Subject: [PATCH 6/7] onnxruntime: cmake: change visibility compilation options

Changing the compile options of the libonnxruntime shared lib
to export all the symbols of the lib and make them visible during
the runtime.
Modifying the CXX compilation flags to disable dangling reference
error showing after compiling with a scarthgap Yocto release.

Signed-off-by: Othmane AHL ZOUAOUI <othmane.ahlzouaoui@st.com>
---
 cmake/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 08a4c6af52..7c992e354e 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -24,6 +24,7 @@ include(CMakeDependentOption)
 include(FetchContent)

 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-dangling-reference")

 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 # NOTE: POSITION INDEPENDENT CODE hurts performance, and it only make sense on POSIX systems
@@ -1620,3 +1621,5 @@ endif()

 message("ONNXRUNTIME_VERSION_MAJOR=${ONNXRUNTIME_VERSION_MAJOR}")
 set_target_properties(onnxruntime PROPERTIES SOVERSION "${ONNXRUNTIME_VERSION_MAJOR}")
+target_compile_options(onnxruntime PRIVATE -fvisibility=hidden)
+target_compile_options(onnxruntime PRIVATE -fvisibility=default)
--
2.34.1

